Rocket.Chat/Controllers/Settings/EditProfileTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+                })
+            } else {
+                self.user = result.user
+                self.isLoading = false
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }, errored: { (_) in
+            Alert(key: "alert.load_profile_error").present(handler: { _ in
+                self.navigationController?.popViewController(animated: true)
+            })
+        })
+    }
+
+    func bindUserData() {
+        DispatchQueue.main.async {
+            self.avatarView.user = self.user
+            self.name.text = self.user?.name
+            self.username.text = self.user?.username
+            self.email.text = self.user?.emails.first?.email
+        }
+    }
+
+    // MARK: Actions
+
+    @IBAction func saveProfile(_ sender: UIBarButtonItem) {
+        hideKeyboard()
+
+        guard let user = user, let userId = user.identifier else { return }
+        guard let name = name.text, let username = username.text, let email = email.text,
+            !name.isEmpty, !username.isEmpty, !email.isEmpty else {
+                Alert(key: "alert.update_profile_empty_fields").present()
+                return
+        }
+
+        user.name = name
+        user.username = username
+        user.emails.first?.email = email
+
+        var password: String?
+
+        if let newPassword = newPassword.text, let passwordConfirmation = passwordConfirmation.text,
+                !newPassword.isEmpty, !passwordConfirmation.isEmpty {
+            if newPassword == passwordConfirmation {
+                password = newPassword
+            } else {
+                Alert(key: "alert.password_mismatch_error").present()
+            }
+        }
+
+        startLoading()
+
+        if let avatarFile = avatarFile {
+            isUploadingAvatar = true
+
+            let client = API.current()?.client(UploadClient.self)
+            client?.uploadAvatar(data: avatarFile.data, filename: avatarFile.name, mimetype: avatarFile.type, completion: {
+                self.isUploadingAvatar = false
+
+                if !self.isUpdatingUser {
+                    self.stopLoading(sender: sender)
+                }
+
+                self.avatarView.shouldRefreshCache = true
+                self.avatarView.avatarPlaceholder = self.avatarView.imageView.image
+                self.avatarView.updateAvatar()
+            })
+        }
+
+        let stopUpdatingUser = { [weak self] in
+            guard let weakSelf = self else { return }
+            weakSelf.isUpdatingUser = false
+            if !weakSelf.isUploadingAvatar {
+                self?.stopLoading(sender: sender)
+            }
+        }
+
+        isUpdatingUser = true
+        let updateUserRequest = UpdateUserRequest(userId: userId, user: user, password: password)
+        api?.fetch(updateUserRequest, succeeded: { result in
+            stopUpdatingUser()
```

Comment:
The call of this method here `stopUpdatingUser()` is because you want to stop the loaders? I think we can improve the name of this method to something like `stopLoading()` or something like that... what do you think?

Reply:
I think that `stopUpdatingUser()` was a bad naming idea, my bad. `stopLoading()` works!